go/def.bzl

```diff
@@ -999,9 +999,172 @@ def cgo_library(name, srcs,
       **kwargs
   )
 
+################
+
+def _go_repository_impl(ctx):
+  fetch_repo = ctx.path(ctx.attr._fetch_repo)
+
+  if ctx.attr.commit and ctx.attr.tag:
+    fail("cannot specify both of commit and tag", "commit")
+  if ctx.attr.commit:
+    rev = ctx.attr.commit
+  elif ctx.attr.tag:
+    rev = ctx.attr.tag
+  else:
+    fail("neither commit or tag is specified", "commit")
+
+  result = ctx.execute([
```

Comment:
Perhaps add a TODO to think about submodules?
https://www.bazel.io/versions/master/docs/be/workspace.html#git_repository.init_submodules